### DISCUSSION TITLE
Gauge is not saving screenshots when running in docker container

### DISCUSSION THREAD

**Discussion by bart-brodz** [getgauge/gauge#2084] [General]
Hi,
I'm running Gauge UI tests in docker using template form:
https://github.com/getgauge/template-js

My tests are executing in docker container, but it seems that gauge has problems with saving them to screenshots output directory.
When running test locally on Windows or Linux screenshots are saved to appropriate directories.

gauge.log:
```[Gauge] [DEBUG] Created gauge_screenshots_dir at /gauge/screenshots
[Gauge] [DEBUG] Plugin json-report is already installed.
[Gauge] [DEBUG] Plugin screenshot is already installed.
[Gauge] [DEBUG] Plugin js is already installed.
... (test specifications running)
[html-report] [DEBUG] Transformed SuiteResult to report structure
[html-report] [WARNING] Failed to read screenhsot open screenshots/screenshot-78645268239785.png: no such file or directory
[html-report] [WARNING] Failed to read screenhsot open screenshots/screenshot-78645555856161.png: no such file or directory
...
html-report] [WARNING] Failed to read screenhsot open screenshots/screenshot-78644974489259.png: no such file or directory
[html-report] [INFO] Successfully generated html-report to => /gauge/reports/html-report/index.html
[html-report] [DEBUG] Done generating HTML report using theme from /home/gauge/.gauge/plugins/html-report/4.1.0/themes/default
[json-report] [INFO] Successfully generated json-report to => /gauge/reports/json-report
```

The problem was occurring even when property `gauge_screenshots_dir = screenshots` wasn't set in `default.properies`, the path in error log was: `/gauge/.gauge/screenshots/screenshot-84371848003397.png`

In `index.html` from `html-report` screenshots are not displaying, because there are no image files in `html-report/images/` directory.
I also exec'd into running container during tests, and neither of these directories contained screenshot files.
I don't have any `gauge.customScreenshotWriter` implementation in code.

Did anybody have similar problem or could point me to what cam be wrong in my configuration?

My configuration:
Dockerfile:
Content is the same as in the template file.
package.json:
```
{
  "name": "gauge-taiko-template",
  "description": "Starter template for writing JavaScript tests for Gauge",
  "scripts": {
    "test": "gauge run specs/"
  },
  "dependencies": {
    "taiko": "1.2.7",
    "clipboardy": "^2.3.0"
  }  
}
```
manifest.json:
```
{
  "Language": "js",
  "Plugins": [
    "html-report",
    "json-report",
    "screenshot"
  ]
}
```
default.properties:
```
gauge_reports_dir = reports
overwrite_reports = true
screenshot_on_failure = true
logs_directory = logs
gauge_specs_dir = specs
csv_delimiter = ,
allow_multiline_step = true
allow_scenario_datatable = true
gauge_screenshots_dir = screenshots
```

**Comment by zabil** [comment]
> I don't have any gauge.customScreenshotWriter implementation in code. 

For headless docker based environment's you must have `gauge.customScreenshotWriter` implementation. Is there any reason you to remove it?

**Reply by bart-brodz** [reply]
Thank you for your reply.
I didn't found any information, that `gauge.customScreenshotWriter` is required for running tests in docker container either in headless mode. Running tests locally on Win or Linux without custom writer worked fine (probably because I had gauge installed).
After adding custom writer screenshots are saved and everything work correctly.
